paymentsheet/src/test/java/com/stripe/android/paymentsheet/FormSaveCheckboxTest.kt

```diff
@@ -125,9 +132,9 @@ class FormSaveCheckboxTest {
                 formArgumentFromSetupIntentTestInput(it.testInput)
             ).isEqualTo(
                 FormFragmentArguments(
-                    SupportedPaymentMethod.fromCode(it.testInput.lpmTypeFormCode)!!.name,
-                    saveForFutureUseInitialVisibility = it.testOutput.expectedSaveCheckboxVisible,
-                    saveForFutureUseInitialValue = it.testOutput.expectedSaveCheckboxValue,
+                    SupportedPaymentMethod.fromCode(it.testInput.lpmTypeFormCode)!!,
+                    allowUserInitiatedReuse = it.testOutput.expectedAllowUserInitiatedReuse,
+                    saveForFutureUseInitialValue = it.testOutput.expectedReusable,
```

Comment:
Just want to double check this - the client should **not** set `setup_future_usage` if the PaymentIntent already has `setup_future_usage` set when it is retrieved.

Put another way, the client should **only** set `setup_future_usage` if its initial value on the PaymentIntent is `nil`.

Reply:
that is correct

Reply:
This is only controlling if the Fields required for reuse are present.  SO for instance if mandate should be shown when it is a setup intent

Reply:
WDYT about replacing these two bools with an enum:

```
enum SaveMode {
  /// We can't save the PaymentMethod. e.g., Payment mode without a customer
  case none

  /// The customer chooses whether or not to save the PaymentMethod. Display a 'save' checkbox.
  case userSelectable

  /// The merchant requires us to save the payment method. Display any mandate text, etc.
  case merchantRequired
}
```

The platforms should have the same modeling and this is what I wrote up for iOS.  I could be convinced to refactor iOS to the two-bool model we have here, though.

Reply:
> Can we rename this argument accordingly then? Like, displayUIRequiredForSaving?
yeah, I will take the refactoring of the naming to displayUIRequiredForSaving in a separate PR, to pull that through the code will be quite a few changes.

> Also, what happens if allowUserInitiatedReuse is true and saveForFutureUseInitialValue is false -- is this like the checkbox is visible but initially deselected?
yeah, if the allowUserInitiatedReuse it true, then it should be setting the saveForFutureUseInitialValue to true, if we want the default checkbox value to be true, which I think we do.



I like the approach of having all the scenarios explicitly written out because it doesn't presume anything about the implementation. I am using it as a bit of a baseline test, and if there is another combination to add it is easy to calculate out all the new combinations and baseline and validate.

Reply:
> yeah, if the allowUserInitiatedReuse it true, then it should be setting the saveForFutureUseInitialValue to true, if we want the default checkbox value to be true, which I think we do.


Is this saying `saveForFutureUseInitialValue` is ignored if `allowUserInitiatedReuse` is true?